src/s3/commands/presignedURL.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SignedUrlRequest } from '../../shared/clients/s3Client'
+import { Env } from '../../shared/vscode/env'
+import { S3FileNode } from '../explorer/s3FileNode'
+import { Window } from '../../shared/vscode/window'
+import { addCodiconToString } from '../../shared/utilities/textUtilities'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { COPY_TO_CLIPBOARD_INFO_TIMEOUT_MS } from '../../shared/constants'
+import { validateTime } from '../util'
+
+export async function presignedURLCommand(
+    node: S3FileNode,
+    window = Window.vscode(),
+    env = Env.vscode()
+): Promise<void> {
+    const time = await promptTime(node.file.key, window)
+    const s3Client = node.s3
+
+    const request: SignedUrlRequest = {
+        bucketName: node.bucket.name,
+        key: node.file.key,
+        operation: 'getObject',
+        time: time,
+    }
+
+    const url = await s3Client.getSignedUrl(request)
+
+    await copyUrl(url, window, env)
+}
+
+export async function promptTime(fileName: string, window = Window.vscode()): Promise<number> {
+    const timeStr = await window.showInputBox({
```

Comment:
What happens if the user inputs a time of '0'? If this creates an unusable URL then we probably should validate for it.

Reply:
Also wondering if there is a max time value allowed by S3. Though if we surface API errors in a reasonable way that should be fine.

Reply:
I couldn't find a max time. However, different credential types have different lengths: https://aws.amazon.com/premiumsupport/knowledge-center/presigned-url-s3-bucket-expiration/

From what I understand from this, if you have a longer period than a credential type allows, it'll revert to the max for that.